docs/development/visualize/development-create-visualization.asciidoc

```diff
@@ -0,0 +1,307 @@
+[[development-create-visualization]]
+=== Developing Visualizations
+
+This is a short description of functions and interfaces provided. For more information you should check the kibana
+source code and the existing visualizations provided with it.
+
+* <<development-react-visualization-type>>
+* <<development-vislib-visualization-type>>
+- <<development-vis-editors>>
+* <<development-default-editor>>
+* <<development-custom-editor>>
+- <<development-visualization-request-handlers>>
+* <<development-default-request-handler>>
+* <<development-none-request-handler>>
+* <<development-custom-request-handler>>
+- <<development-visualization-response-handlers>>
+* <<development-default-response-handler>>
+* <<development-none-response-handler>>
+* <<development-custom-response-handler>>
+- <<development-vis-object>>
+* <<development-vis-timefilter>>
+
+[[development-visualization-factory]]
+=== Visualization Factory
+
+To create a new visualization you need to create new visualization type using the VisualizationFactory. 
+The factory can create `Base`, `Angular`, `React` and `Vislib` visualization.
+
+["source","html"]
+-----------
+import { VisFactoryProvider } from 'ui/vis/vis_factory';
+
+export default function MyNewVisType(Private) {
+  const VisFactory = Private(VisFactoryProvider);
+
+  return  VisFactory.createBaseVisualization({
+    name: 'my_new_vis',
+    title: 'My New Vis',
+    icon: 'my_icon',
+    description: 'Cool new chart',
+    ...
+  });
+}
+-----------
+
+The list of common parameters:
+
+- *name*: unique visualization name, only lowercase letters and underscore
+- *title*: title of your visualization as displayed in kibana
+- *icon*: the icon class to use (font awesome)
+- *image*: instead of icon you can provide svg image (imported)
+- *description*: description of your visualization as shown in kibana
+- *category*: the category your visualization falls into (one of `ui/vis/vis_category` values)
+- *visConfig*: object holding visualization parameters
+- *visConfig.defaults*: object holding default visualization configuration
+- *visualization*: A constructor function for a Visualization.
+- *requestHandler*: <string> one of the available request handlers or <function> for custom one
+- *responseHandler*: <string> one of the available response handlers or <function> for custom one
+- *editor*: <string> one of the available editors or Editor class for custom one
+- *editorConfig*: object holding editor parameters
+- *options.showTimePicker*: <bool> show or hide time picker (defaults to true)
+- *options.showQueryBar*: <bool> show or hide query bar (defaults to true)
+- *options.showFilterBar*: <bool> show or hide filter bar (defaults to true)
+- *options.showIndexSelection*: <bool> show or hide index selection (defaults to true)
+- *options.hierarchicalData*: <bool> use hierarchical data (defaults to false) [BWC]
```

Comment:
i removed this as its there for backward compatibility reasons and plugin developers should not actually be using it.